ctci/chapter2/2.1.py

```diff
@@ -0,0 +1,65 @@
+#2.1
+
+#Remove Dups: Write code to remove duplicates from an unsorted li nked list. 
+#FOLLOW UP
+#How would you solve this problem if a temporary buffer is not allowed?
+#   n^2 sollution? don't see a better way, might check hint later
+#   apperntly you can inplace merge sort in O(nlog(n)) 
+#   then you can remove duplicates in one pass, so better than n^2
+
+
+#assuming no cycles
+from random import randrange
+
+class Node:
+    def __init__(self, dataval=None):
+        self.dataval = dataval
+        self.nextval = None
+
+class SLinkedList:
+    def __init__(self):
+        self.headval = None
+
+
+def remDup (h: SLinkedList)-> SLinkedList:
+    s = set()
+    i = h.headval
+    p = h.headval
+    while (i != None):
+        if (i.dataval in s):
+            if (i == h.headval):
```

Comment:
whats the point of this? The head is never not unique given its the first node you visit